apis/python/src/tiledbsoma/options/_soma_tiledb_context.py

```diff
@@ -118,6 +120,16 @@ def __init__(
                 provided, a new ThreadPoolExecutor will be created with
                 default settings.
         """
+        if tiledb_ctx is not None:
+            version = get_implementation_version().split(".")
+            assert (int(version[0]), int(version[1])) < (1, 14)
+            warnings.warn(
+                "The tiledb_ctx parameter will be deprecated in 1.14. "
+                "Use tiledb_config instead by passing "
+                "SOMATileDBContext(tiledb_config=ctx.config()).",
+                DeprecationWarning,
```

Comment:
I’m a bit confused by the language here. If the parameter isn’t deprecated yet, then what is the warning for? If the intent is to *remove* it in v1.14, then I would say it’s deprecated _now_ for _removal_ in v1.14. As it stands, I would read “will be deprecated in v1.14” to mean that it works now, it will continue to work in v1.14 (but be even more deprecated somehow?), and at some indeterminate point after v1.14, it will stop working.

Reply:
> deprecated now for removal in v1.14

+1

FWIW I had several co-workers at a former job who (incorrectly) used "deprecate" to mean "get rid of". And I suspect they're not alone; others may have a similar misread of what "deprecate" means. So it's good to be extra-clear on what we mean: "deprecating now" means "flagging that people should stop using this", and "removal in 1.14" meaning that's when it will no longer exist.

Reply:
Gotcha - thanks for the clarification. I was conflating deprecation and removal and will clear up the language here.